src/inet/arpa-inet-compatibility.h

```diff
@@ -49,6 +56,26 @@
 
 #if CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT
 
+#ifndef BYTE_ORDER
+#if defined(__LITTLE_ENDIAN__)
+#define BYTE_ORDER LITTLE_ENDIAN
```

Comment:
This does not work if LITTLE_ENDIAN is not defined right? In the new GCC case, neither LITTLE_ENDIAN nor BIG_ENDIAN nor BYTE_ORDER are defined.

lets not try to use BYTE_ORDER at all ... it does not seem standard.

Reply:
Quote what I mentioned on slack

> from what I understood, different scenarios with different possibles macros
> In the case we're __LITTLE_ENDIAN__ or __BIG_ENDIAN__ their counterparts without then__ prefix/suffix will also be defined with a set value
> In the case they are not and __BYTE_ORDER__ is defined then __ORDER_LITTLE_ENDIAN__  should also be defined. That's why I do #define LITTLE_ENDIAN __ORDER_LITTLE_ENDIAN__ after.

But for simplicity ill remove the other condition and BYTE_ORDER all together and only use `__BYTE_ORDER__`

If there is a case or platform that do not have this define then at least the #ifndef error will prevent the build and notify the developer (and CI should catch it)